tests/testthat/test-rmd_to_md.R

```diff
@@ -1,100 +1,69 @@
+
+  rmd_to_md(
+    rmd_file = testthat::test_path("example.Rmd"),
+    md_dir = expected_md_dir,
+    fig_dir = expected_fig_dir,
+    fig_url_dir = "/software/example/",
+    sidebar_order = 2
+  )
 
   expect_identical(
-    list.files(md_dir),
-    c("occurrence-process.md")
+    list.files(expected_md_dir),
+    c("example.md")
+  )
+  expect_identical(
+    list.files(expected_fig_dir),
+    c("example-unnamed-chunk-3-1.png")
   )
-
-  unlink(temp_dir, recursive = TRUE)
 })
 
-test_that("rmd_to_md() writes figures to a directory", {
-  skip("Test broken see issue #2")
-  skip_if_offline()
-  rmd_file <- "https://raw.githubusercontent.com/b-cubed-eu/gcube/refs/heads/main/vignettes/articles/occurrence-process.Rmd"
-
+test_that("rmd_to_md() writes the expected markdown, including custom
+           frontmatter and figure paths", {
   temp_dir <- tempdir()
-
-  md_dir <- file.path(temp_dir, "src", "content", "docs", "r", "gcube")
-  fig_dir <- file.path(temp_dir, "public", "r", "gcube")
-  fig_url_dir <- paste0(temp_dir, "/astro-docs/r/gcube/")
-  order <- 1
-
-  rmd_to_md(rmd_file, md_dir, fig_dir, fig_url_dir, order)
-
-  expect_equal(
-    list.files(fig_dir),
-    c(
-      "occurrence-process-unnamed-chunk-12-1.png",
-      "occurrence-process-unnamed-chunk-3-1.png",
-      "occurrence-process-unnamed-chunk-7-1.png",
-      "occurrence-process-unnamed-chunk-9-1.png"
-    )
+  expected_md_dir <- file.path(temp_dir, "src/content/docs/software/example")
+  on.exit(unlink(temp_dir, recursive = TRUE))
+
+  rmd_to_md(
+    rmd_file = testthat::test_path("example.Rmd"),
+    md_dir = expected_md_dir,
+    fig_dir = file.path(temp_dir, "public/software/example"),
+    fig_url_dir = "/software/example/",
+    title = "Custom title",
+    sidebar_label = "Custom sidebar label",
+    sidebar_order = 2
   )
 
-  unlink(temp_dir, recursive = TRUE)
+  expect_snapshot_file(
+    file.path(expected_md_dir, "example.md"),
+    transform = function(x) {
+      gsub(format(Sys.time(), "%Y-%m-%d"), "<current_date>", x)
+    }
+  )
 })
 
 test_that("rmd_to_md() resets knitting options to the original settings", {
-  skip_if_offline()
-  rmd_file <- "https://raw.githubusercontent.com/b-cubed-eu/gcube/refs/heads/main/vignettes/articles/occurrence-process.Rmd"
-
   temp_dir <- tempdir()
-
-  md_dir <- file.path(temp_dir, "src", "content", "docs", "r", "gcube")
-  fig_dir <- file.path(temp_dir, "public", "r", "gcube")
-  fig_url_dir <- paste0(temp_dir, "/astro-docs/r/gcube/")
-  order <- 1
-
+  on.exit(unlink(temp_dir, recursive = TRUE))
   original_opts_knit <- knitr::opts_knit$get()
 
-  rmd_to_md(rmd_file, md_dir, fig_dir, fig_url_dir, order)
-
+  rmd_to_md(
+    rmd_file = testthat::test_path("example.Rmd"),
+    md_dir = file.path(temp_dir, "src/content/docs/software/example"),
+    fig_dir = file.path(temp_dir, "public/software/example"),
+    fig_url_dir = "/software/example/",
+    sidebar_order = 2
```

Comment:
Please remove this line (and comma above) once `sidebar_order` is no longer required.